### ISSUE TITLE
[speech-to-text] SpeechToText.ParseRecognizeResponse fails to set result_index

### ISSUE THREAD

**Issue by JoeStrout** [watson-developer-cloud/unity-sdk#347]
The SpeechRecognitionEvent class has a result_index property, and the JSON data from the server that we're parsing in SpeechToText.ParseRecognizeResponse includes a "result_index" property.  But the code never actually sets the former from the latter.

As a work-around, I added this code near the bottom, right before the catch-block (in place of the current "return new SpeechRecognitionEvent..." line):

```
	            var srEvent = new SpeechRecognitionEvent(results.ToArray());
	            int foo;
	            if (int.TryParse(resp["result_index"].ToString(), out foo)) srEvent.result_index = foo;
	            return srEvent;
```

**Comment by mediumTaj** [comment]
Good catch! Please see this PR https://github.com/watson-developer-cloud/unity-sdk/pull/349
